.github/workflows/merge-to-stage.js

```diff
@@ -211,11 +211,25 @@ const openStageToMainPR = async () => {
 
 const mergeLimitExceeded = () => MAX_MERGES - existingPRCount < 0;
 
+const isBeforeRCP = () => {
+  const now = new Date();
+  const { RCPDates } = require('./helpers.js');
+  
+  return RCPDates.some(({ start }) => {
+    const rcpStart = new Date(start);
+    const hoursUntilRCP = (rcpStart - now) / (1000 * 60 * 60);
+    const fourDaysBeforeRCP = 96;
+    return hoursUntilRCP > 0 && hoursUntilRCP <= fourDaysBeforeRCP;
+  });
+};
+
 const main = async (params) => {
   github = params.github;
   owner = params.context.repo.owner;
   repo = params.context.repo.repo;
   if (isWithinRCP({ offset: process.env.STAGE_RCP_OFFSET_DAYS || 2, excludeShortRCP: true })) return console.log('Stopped, within RCP period.');
+  if (isBeforeRCP()) return console.log('Stopped, within 4 days before RCP to have a clean stage branch for emergencies.');
```

Comment:
Shouldn't this also consider the short RCPs use-case? We sometimes have RCPs that last a few hours, the batches shouldn't be blocked for 4 days before those. The conditional above uses a `excludeShortRCP` flag; not sure if it can be re-used, but might be worth checking out.